src/layer/ImageOverlay.js

```diff
@@ -165,6 +169,7 @@ export var ImageOverlay = Layer.extend({
 		img.onmousemove = Util.falseFn;
 
 		img.onload = Util.bind(this.fire, this, 'load');
+		img.onerror = Util.bind(this._overlayOnError, this, 'error');

```

Comment:
To avoid the pretty weird case where the error image itself 404s. `TileLayer` has the same logic, and I think this is more or less a copy of that.